crates/ruff_linter/src/rules/pyupgrade/snapshots/ruff_linter__rules__pyupgrade__tests__UP050.py.snap

```diff
@@ -0,0 +1,322 @@
+   35 |+class C[**P, **_P1]: ...
+36 36 | class C[**P](Generic[_P2]): ...
+37 37 | class C[**P](Generic[_P3]): ...
+38 38 | 
+
+UP050.py:36:14: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+35 | class C[**P](Generic[_P1]): ...
+36 | class C[**P](Generic[_P2]): ...
+   |              ^^^^^^^^^^^^ UP050
+37 | class C[**P](Generic[_P3]): ...
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+33 33 | 
+34 34 | 
+35 35 | class C[**P](Generic[_P1]): ...
+36    |-class C[**P](Generic[_P2]): ...
+   36 |+class C[**P, **_P2]: ...
+37 37 | class C[**P](Generic[_P3]): ...
+38 38 | 
+39 39 | 
+
+UP050.py:37:14: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+35 | class C[**P](Generic[_P1]): ...
+36 | class C[**P](Generic[_P2]): ...
+37 | class C[**P](Generic[_P3]): ...
+   |              ^^^^^^^^^^^^ UP050
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+34 34 | 
+35 35 | class C[**P](Generic[_P1]): ...
+36 36 | class C[**P](Generic[_P2]): ...
+37    |-class C[**P](Generic[_P3]): ...
+   37 |+class C[**P, **_P3]: ...
+38 38 | 
+39 39 | 
+40 40 | class C[T](Generic[T, _A]): ...
+
+UP050.py:40:12: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+40 | class C[T](Generic[T, _A]): ...
+   |            ^^^^^^^^^^^^^^ UP050
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+37 37 | class C[**P](Generic[_P3]): ...
+38 38 | 
+39 39 | 
+40    |-class C[T](Generic[T, _A]): ...
+   40 |+class C[T, _A]: ...
+41 41 | 
+42 42 | 
+43 43 | # See `is_existing_param_of_same_class`
+
+UP050.py:46:35: UP050 Class with type parameter list inherits from `Generic`
+   |
+44 | # `expr_name_to_type_var` doesn't handle named expressions,
+45 | # only simple assignments, so there is no fix.
+46 | class C[T: (_Z := TypeVar('_Z'))](Generic[_Z]): ...
+   |                                   ^^^^^^^^^^^ UP050
+   |
+   = help: Convert to new-style
+
+UP050.py:50:16: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+49 | class C(Generic[_B]):
+50 |     class D[T](Generic[_B, T]): ...
+   |                ^^^^^^^^^^^^^^ UP050
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+47 47 | 
+48 48 | 
+49 49 | class C(Generic[_B]):
+50    |-    class D[T](Generic[_B, T]): ...
+   50 |+    class D[T, _B: int]: ...
+51 51 | 
+52 52 | 
+53 53 | class C[T]:
+
+UP050.py:54:16: UP050 Class with type parameter list inherits from `Generic`
+   |
+53 | class C[T]:
+54 |     class D[U](Generic[T, U]): ...
+   |                ^^^^^^^^^^^^^ UP050
+   |
+   = help: Convert to new-style
+
+UP050.py:57:12: UP050 [*] Class with type parameter list inherits from `Generic`
+   |
+57 | class C[T](Generic[_C], Generic[_D]): ...
+   |            ^^^^^^^^^^^ UP050
+   |
+   = help: Convert to new-style
+
+ℹ Unsafe fix
+54 54 |     class D[U](Generic[T, U]): ...
+55 55 | 
+56 56 | 
+57    |-class C[T](Generic[_C], Generic[_D]): ...
+   57 |+class C[T, _C: (str, bytes)](Generic[_D]): ...
```

Comment:
...fair enough 😆

I guess if you were actually running Ruff from the command line with `--unsafe-fixes --fix`, it would get rid of both the `Generic` base classes, because it runs all selected rules in a loop until there are no more autofixes left to apply? If so, could you maybe add a comment to that effect in the fixture file?